cmd/cdi-importer/importer.go

```diff
@@ -190,12 +190,14 @@ func main() {
 		if err != nil {
 			klog.Errorf("%+v", err)
 			if err == importer.ErrRequiresScratchSpace {
+				dp.Close()
```

Comment:
Why is this needed if we already have `defer dp.Close()` up above?

Reply:
Apparently runtime.Goexit exists to be os.Exit while running deferred calls.

Reply:
Thanks, that would be much cleaner. It looks like it's meant to be called from a goroutine though, I will run some tests and see how well it works here.

Reply:
Goexit does not appear to be a drop-in replacement, it hangs after the transfer if I just use it in place of os.Exit. It does unlock the disk though, so it must be running the deferred Close correctly.

Reply:
thanks for trying